src/utils/osUtils.ts

```diff
@@ -88,3 +90,9 @@ export async function getDockerOSType(context: IActionContext): Promise<DockerOS
         return info.OSType;
     }
 }
+
+export async function getConatinerOSType(containerId: string): Promise<ContainerOSType> {
+    const inspectCmd: string = `docker inspect --format="{{ .Platform}}" ${containerId}`;
+    const execResult = await execAsync(inspectCmd);
```

Comment:
Should be able to use Dockerode instead which will be much faster.

Reply:
Dockerode didn't provide this property and moreover it require to construct the the container object, then query it.

Reply:
"Constructing" the container object does not involve any HTTP requests and only requires the ID, so that's not an issue. There's probably another property we can use to figure it out.

Reply:
The property is there, `.Platform` as expected.

Reply:
Are you suggesting to fix the dockerode package?

Reply:
Just `@types/dockerode`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/dockerode

Reply:
Made a PR to update the types/dockerode. Once it is merged and a new version is available, I will consume here.